shopify_python/git_utils.py

```diff
@@ -22,10 +22,14 @@ def _remote_origin_master(git_repo):
 
 
 def _modified_in_branch(git_repo, other_ref):
-    # type: (repo.Repo, head.Head) -> typing.List[str]
-    commit = git_repo.active_branch.commit
-    modified = [x for x in commit.diff(other_ref.commit, R=True) if not x.deleted_file]
-    return [x.b_path for x in modified]
+    # type: (repo.Repo, head.Head) -> typing.Sequence[str]
+    common_commit = git_repo.merge_base(git_repo.active_branch, other_ref)[0]
```

Comment:
What if there is no merge base? Maybe return a more descriptive exception than `IndexError`.